transport/transport.test.ts

```diff
@@ -536,6 +538,165 @@ describe.each(testMatrix())(
   },
 );
 
+describe.each(testMatrix())(
+  'transport disabling transparent reconnect ($transport.name transport, $codec.name codec)',
+  async ({ transport, codec }) => {
+    const opts: TransportOptions = {
+      codec: codec.codec,
+      heartbeatIntervalMs: 1_000,
+      heartbeatsUntilDead: 2,
+      // set the session disconnect grace to 0 to force a hard reconnect
+      sessionDisconnectGraceMs: 0,
+      handshakeTimeoutMs: 5_000,
+    };
+    let testHelpers: TestSetupHelpers;
+    let getClientTransport: TestSetupHelpers['getClientTransport'];
+    let getServerTransport: TestSetupHelpers['getServerTransport'];
+    const { addPostTestCleanup, postTestCleanup } = createPostTestCleanups();
+    beforeEach(async () => {
+      testHelpers = await transport.setup({ client: opts, server: opts });
+      getClientTransport = testHelpers.getClientTransport;
+      getServerTransport = testHelpers.getServerTransport;
+      return async () => {
+        await postTestCleanup();
+        await testHelpers.cleanup();
+      };
+    });
+
+    test('reconnecting with grace period of 0 should result in hard reconnect', async () => {
+      const clientTransport = getClientTransport('client');
+      const serverTransport = getServerTransport();
+      addPostTestCleanup(async () => {
+        await cleanupTransports([clientTransport, serverTransport]);
+      });
+
+      await waitFor(() => expect(clientTransport.connections.size).toEqual(1));
+      await waitFor(() => expect(serverTransport.connections.size).toEqual(1));
+
+      const oldClientSession = serverTransport.sessions.get('client');
+      const oldServerSession = clientTransport.sessions.get('SERVER');
+      expect(oldClientSession).toMatchObject({
+        id: oldServerSession?.advertisedSessionId,
+      });
```

Comment:
```suggestion
      expect(oldClientSession.id).toBe(oldServerSession?.advertisedSessionId);
```

seems to read slightly better

Reply:
reverting this due to a lint warning:
`transport/transport.test.ts:578:14 - error TS18048: 'oldClientSession' is possibly 'undefined'.`